src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -1291,6 +1366,401 @@ std::shared_ptr<chipstar::Event> CHIPQueueOpenCL::enqueueBarrierImpl(
   return Event;
 }
 
+/********************************************************************************/
+
+chipstar::GraphNative *CHIPQueueOpenCL::createNativeGraph() {
+  // should not raise an error if we fail to create a graph,
+  // because there is a fallback solution
+  CHIPContextOpenCL *Ctx = (CHIPContextOpenCL *)ChipContext_;
+  if (!Ctx->supportsCommandBuffers())
+    return nullptr;
+
+  cl_command_queue CQ = ClQueue_->get();
+  int err = CL_SUCCESS;
+  cl_command_buffer_khr Res =
+      Ctx->exts()->clCreateCommandBufferKHR(1, &CQ, 0, &err);
+  if (Res == nullptr || err != CL_SUCCESS) {
+    logError("clCreateCommandBufferKHR FAILED with status {}",
+             resultToString(err));
+    return nullptr;
+  }
+
+  return new CHIPGraphNativeOpenCL(Res, CQ, Ctx->exts());
+}
+
+std::shared_ptr<chipstar::Event>
+CHIPQueueOpenCL::enqueueNativeGraph(chipstar::GraphNative *NativeGraph) {
+
+  std::shared_ptr<chipstar::Event> Event =
+      ::Backend->createCHIPEvent(ChipContext_);
+  CHIPContextOpenCL *Ctx = (CHIPContextOpenCL *)ChipContext_;
+  CHIPGraphNativeOpenCL *G = (CHIPGraphNativeOpenCL *)NativeGraph;
+  if (!Ctx->supportsCommandBuffers())
+    return nullptr;
+  if (NativeGraph == nullptr)
+    return nullptr;
+  cl_command_queue CQ = ClQueue_->get();
+  int Status = Ctx->exts()->clEnqueueCommandBufferKHR(
+      1, &CQ, G->get(), 0, nullptr,
+      std::static_pointer_cast<CHIPEventOpenCL>(Event)->getNativePtr());
+  CHIPERR_CHECK_LOG_AND_THROW(Status, CL_SUCCESS, hipErrorTbd);
+
+  return Event;
+}
+
+void CHIPQueueOpenCL::destroyNativeGraph(chipstar::GraphNative *NativeGraph) {
+  if (NativeGraph == nullptr)
+    return;
+  CHIPGraphNativeOpenCL *G = (CHIPGraphNativeOpenCL *)NativeGraph;
+  delete G;
+}
+
+bool CHIPGraphNativeOpenCL::addNode(chipstar::GraphNode *NewNode) {
```

Comment:
Should these be OpenCLQueue and OpenCLNativeGraph? With CHIP dropped and changed to the chipstar namespace?

Reply:
Maybe if it's decided one day to rename all backend classes, but currently they are all called CHIPxyzOpenCL and have no namespace, so this is consistent with current status.

Reply:
I think we decided to indeed do that. If it's only partially done, we should convert the nonconverted ones, not add new wrongly named ones to away from that decision, right?